_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,82 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The aim of this project is to automatically create whitelists (to be utilized
+by vulture) for projects using sip files to generate Python bindings for C++
+code, most significant of them being PyQt. Once implemented, a whitelist
+generated through this generator for PyQt could be universally utilised across
+all PyQt projects.
+
+#### IDEAS
+
+* Create a command line tool which takes in sip files as input and spits a
+  whitelist with all the virtual functions defined in the sip files. (Add
+  Documentation and tests)
+* Use this tool to generate a whitelist for PyQt and ship it with vulture. This
+  will reduce overhead for PyQt projects which use vulture.
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding
+  generator used for PySide2
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `xml`
+  output from coverage.py
+* Get familiar with coverage.xml
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst)
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/28)
+
+##### CODING PHASE 1
+* The whitelist generator should be completely implemented.
+* It should reside in it's own repository, with CI enabled for testing on both
```

Comment:
this repo/project needs to maintain 100% test coverage.

very likely this will be part of coala , if not bears, it would need to be part of the `coala-quickstart` program (a project for 2019!) to generate the whitelist for VultureBear and maybe other bears also.

Reply:
This comment has been addressed below.